src/tss2-esys/esys_tr.c

```diff
@@ -392,7 +394,20 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
         if (authValue->size > sizeof(TPMU_HA)) {
             return_error(TSS2_ESYS_RC_BAD_SIZE, "Bad size for auth value.");
         }
```

Comment:
Shouldn't this condition go away, because now we hash it?

Reply:
No the maximal size still will be sizeof(TMU_HA).

Reply:
Only when you're not hashing because the resource doesn't have a name alg. All nameAlg digest sizes will be less then TPMU_HA by design.

I think the code needs to be:
```diff
diff --git a/src/tss2-esys/esys_tr.c b/src/tss2-esys/esys_tr.c
index 33d75fd10ad8..73a891c8bfd6 100644
--- a/src/tss2-esys/esys_tr.c
+++ b/src/tss2-esys/esys_tr.c
@@ -391,9 +391,6 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
     if (authValue == NULL) {
         esys_object->auth.size = 0;
     } else {
-        if (authValue->size > sizeof(TPMU_HA)) {
-            return_error(TSS2_ESYS_RC_BAD_SIZE, "Bad size for auth value.");
-        }
         /* Determine name alg of resource */
         if (esys_object->rsrc.rsrcType == IESYSC_KEY_RSRC) {
             name_alg = esys_object->rsrc.misc.rsrc_key_pub.publicArea.nameAlg;
@@ -402,11 +399,15 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
         } else {
             name_alg = TPM2_ALG_NULL;
         }
-        esys_object->auth = *authValue;
         /* Adapt auth value to hash for large auth values. */
         if (name_alg != TPM2_ALG_NULL) {
             r = iesys_hash_long_auth_values(&esys_object->auth, name_alg);
             return_if_error(r, "Adaption of out value failed.");
+        } else {
+            if (authValue->size > sizeof(TPMU_HA)) {
+                return_error(TSS2_ESYS_RC_BAD_SIZE, "Bad size for auth value.");
+            }
+            esys_object->auth = *authValue;
         }
     }
     return TSS2_RC_SUCCESS;
```

Reply:
Disregard this comment, duh moment, we're bounded by the incoming TPM2B_AUTH, whic is TPMU_HA sized buffer anyways.